mlflow/tracing/provider.py

```diff
@@ -128,6 +126,18 @@ def enable():
     _setup_tracer_provider()
 
 
+@contextlib.contextmanager
+def disabled():
+    """
+    Temporarily disable tracing for the duration of the context manager.
+    """
+    try:
+        disable()
+        yield
+    finally:
+        enable()
```

Comment:
What if tracing was already disabled?

Reply:
Ooh you're right, great catch! Will add tests for this, too.